aten/src/ATen/native/mkldnn/Utils.h

```diff
@@ -74,6 +74,20 @@ inline bool mkldnn_bf16_device_check() {
      && cpuinfo_has_x86_avx512vl() && cpuinfo_has_x86_avx512dq()) || (cpuinfo_has_arm_bf16()));
 }
 
+inline bool mkldnn_fp16_device_check() {
+  return cpuinfo_initialize() && cpuinfo_has_x86_avx512fp16();
+}
+
+inline void mkldnn_check_low_precision(ScalarType input_t){
+  if (input_t == ScalarType::BFloat16) {
+    TORCH_CHECK(mkldnn_bf16_device_check(),
+        "mkldnn_convolution_transpose: bf16 path needs the cpu support avx512bw, avx512vl and avx512dq");
+  } else if (input_t  == ScalarType::Half) {
+    TORCH_CHECK(mkldnn_fp16_device_check(),
+        "mkldnn_convolution_transpose: fp16 path needs the cpu support avx512fp16");
```

Comment:
avx512fp16 doesn't have dot-product support. Are we sure that onednn has efficient conv/gemm support based on that?

Reply:
we need some benchmark number here.

Reply:
> avx512fp16 doesn't have dot-product support. Are we sure that onednn has efficient conv/gemm support based on that?

Sorry for slow reply. I'm occupied by a urgent task. The performance should be worse than fp32 for computation bound ops on avx512fp16 platforms since there is internal fp16<->fp32 overhead.  Avx512_fp16 has compute instructions (fma, etc) , but those accumulate to f16, not f32. So unfortunately oneDNN cant use avx512_f16 compute instructions, only convert instructions are used.

Reply:
So, if this is only for the reference implementation, not necessarily we would rely on avx512fp16, right? Also, with avx512fp16, there might be accuracy concern since the accumulation is with f16. We shouldn't turn it on by default, right?

Reply:
I think we would rely on avx512_fp16 since OneDNN requires the cpu has avx512_fp16 at least. Onednn does not use avx512_f16 to compute.  Only convert instructions are used to prevent accuracy issue.

Reply:
But why has it to depend on avx512_fp16 if only conversion instructions are used for ref implementation? Can f16c be used instead?

Reply:
Currently oneDNN has no interest in supporting fp16 for anything earlier platforms than SPR so it requires the cpu has avx512_fp16 at least. oneDNN fp16 implements have much better performance compared to naive fp16 implements of PyTorch (shapes collect from RN50, not adequate testing) https://github.com/pytorch/pytorch/pull/99497#issue-1674040183 .  It benefits to call oneDNN with avx512_fp16.  We can add some conditions to avoid some slow cases of oneDNN, and ask onednn to improve the performance for such cases.